internal/supervisor/installation_deletion.go

```diff
@@ -66,7 +72,18 @@ func (s *InstallationDeletionSupervisor) Do() error {
 		return nil
 	}
 
+	status, err := s.store.GetInstallationsStatus()
+	if err != nil {
+		s.logger.WithError(err).Warn("Failed to query for installation status")
+		return nil
+	}
+	s.currentlyUpdatingCounter = status.InstallationsUpdating
+
 	for _, installation := range installations {
+		if s.currentlyUpdatingCounter >= s.currentlyUpdatingLimit {
+			s.logger.Infof("Max installation updating counter (%d) reached for pending deletions", s.currentlyUpdatingLimit)
+			return nil
+		}
```

Comment:
Not sure if it is the best way of doing this. Given that there are several provisioners changing installations state this limit is kind of raw estimation rather than the hard limit.

The number of processing installations can change between Supervising different installations therefore if we say there was 20 installations queried by the `Do` method, by the time we get to the 20th the number of updating installations could have changed significantly so we might be over the limit.

It might not be a huge issue but it is kind of unexpected behaviour so if we are fine with it, we should document it clearly, both with comments and in flag description.

Reply:
Yep, I was aware of this being a simple soft limit, but it's definitely a good idea to clarify. Added a chunk of comments to call out this point. Thanks!